lib/class-wp-rest-plugins-controller.php

```diff
@@ -0,0 +1,829 @@
+ * Block Directory REST API: WP_REST_Plugins_Controller class
+ *
+ * @since   6.5.0
+ * @package gutenberg
+ */
+
+/**
+ * Core class to access plugins via the REST API.
+ *
+ * @since 5.5.0
+ *
+ * @see   WP_REST_Controller
+ */
+class WP_REST_Plugins_Controller extends WP_REST_Controller {
+
+	const PATTERN = '[^.\/]+(?:\/[^.\/]+)?';
+
+	/**
+	 * Plugins controller constructor.
+	 *
+	 * @since 5.5.0
+	 */
+	public function __construct() {
+		$this->namespace = '__experimental';
+		$this->rest_base = 'plugins';
+	}
+
+	/**
+	 * Registers the routes for the plugins controller.
+	 *
+	 * @since 5.5.0
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_items' ),
+					'permission_callback' => array( $this, 'get_items_permissions_check' ),
+					'args'                => $this->get_collection_params(),
+				),
+				array(
+					'methods'             => WP_REST_Server::CREATABLE,
+					'callback'            => array( $this, 'create_item' ),
+					'permission_callback' => array( $this, 'create_item_permissions_check' ),
+					'args'                => array(
+						'slug'   => array(
+							'type'        => 'string',
+							'required'    => true,
+							'description' => __( 'WordPress.org plugin directory slug.', 'gutenberg' ),
+						),
+						'status' => array(
+							'description' => __( 'The plugin activation status.', 'gutenberg' ),
+							'type'        => 'string',
+							'enum'        => is_multisite() ? array( 'inactive', 'active', 'network-active' ) : array( 'inactive', 'active' ),
+							'default'     => 'inactive',
+						),
+					),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
+			)
+		);
+
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/(?P<plugin>' . self::PATTERN . ')',
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_item' ),
+					'permission_callback' => array( $this, 'get_item_permissions_check' ),
+				),
+				array(
+					'methods'             => WP_REST_Server::EDITABLE,
+					'callback'            => array( $this, 'update_item' ),
+					'permission_callback' => array( $this, 'update_item_permissions_check' ),
+					'args'                => $this->get_endpoint_args_for_item_schema( WP_REST_Server::EDITABLE ),
+				),
+				array(
+					'methods'             => WP_REST_Server::DELETABLE,
+					'callback'            => array( $this, 'delete_item' ),
+					'permission_callback' => array( $this, 'delete_item_permissions_check' ),
+				),
+				'args'   => array(
+					'context' => $this->get_context_param( array( 'default' => 'view' ) ),
+					'plugin'  => array(
+						'type'              => 'string',
+						'pattern'           => self::PATTERN,
+						'validate_callback' => static function ( $file ) {
```

Comment:
Let's make this methods and not static functions. It is improve testabilitiy.

Reply:
Fixed in f525a35.